pkg/virt-handler/vm.go

```diff
@@ -955,13 +955,15 @@ func (d *VirtualMachineController) updateGuestAgentConditions(vmi *v1.VirtualMac
 		}
 
 		var supported = false
+		var reason = ""
 
 		// For current versions, virt-launcher's supported commands will always contain data.
 		// For backwards compatibility: during upgrade from a previous version of KubeVirt,
 		// virt-launcher might not provide any supported commands. If the list of supported
 		// commands is empty, fall back to previous behavior.
 		if len(guestInfo.SupportedCommands) > 0 {
-			supported = isGuestAgentSupported(vmi, guestInfo.SupportedCommands)
+			supported, reason = isGuestAgentSupported(vmi, guestInfo.SupportedCommands)
+			log.Log.V(3).Object(vmi).Info(reason)
```

Comment:
I am not sure about moving the log here. I like that is the function itself that prints out the output of its computation, also for future uses; if some other part will use that function it must log the reason too. You can always get the returned value and using it in the `VirtualMachineInstanceCondition`. Also, you can return the reason only and only if the command is not supported. WDYT?

Reply:
Sorry, I only noticed now that my comment is the opposite of https://github.com/kubevirt/kubevirt/pull/7811#discussion_r882617537
Feel free to adopt the solution you prefer, thanks!